File: 05-Recursion/Test/src/ArrayStack.java
public class ArrayStack<E> implements Stack<E> {

    private Array<E> arr;

    public ArrayStack(){
        arr = new Array<>();
    }

    @Override
    public int getSize(){
        return arr.getSize();
    }

    @Override
    public boolean isEmpty(){
        return arr.isEmpty();
    }

    @Override
    public void push(E e){
        arr.addLast(e);
    }

    @Override
    public E pop(){
        return arr.removeLast();
    }

    @Override
    public E peek(){
        return arr.getLast();
    }



    @Override
    public String toString(){
        StringBuilder res = new StringBuilder();
        res.append("Stack: ");
        res.append("[");
        for (int i=0; i<arr.getSize(); i++){
            res.append(arr.get(i));
            if (i != arr.getSize() - 1)
                res.append(", ");
        }
        res.append("] top");
        return res.toString();
    }

    public static void main(String[] args) {

        ArrayStack<Integer> stack = new ArrayStack<>();

        for (int i=0; i<5; i++) {
            stack.push(i);
            System.out.println(stack);
        }

        stack.pop();
        System.out.println(stack);

        int n = stack.peek();
        System.out.println("栈顶元素值："+ n);
    }
}

File: go/03-Stacks-and-Queues/02-Array-Stack/arraystack/arraystack.go
package arraystack

import (
	"bytes"
	"fmt"
	"github.com/dingsw2019/play-with-data-struct/go/03-Stacks-and-Queues/02-Array-Stack/array"
)

type ArrayStack struct {
	array *array.Array
}

func New(capacity int) *ArrayStack {
	return &ArrayStack{
		array: array.New(capacity),
	}
}

func (as *ArrayStack) GetSize() int {
	return as.array.GetSize()
}

func (as *ArrayStack) IsEmpty() bool {
	return as.array.IsEmpty()
}

// 入栈
func (as *ArrayStack) Push(e interface{}) {
	as.array.AddLast(e)
}

// 出栈
func (as *ArrayStack) Pop() interface{} {
	return as.array.RemoveLast()
}

// 查看栈顶值
func (as *ArrayStack) Peek() interface{} {
	return as.array.GetLast()
}

func (as *ArrayStack) String() string {
	var buf bytes.Buffer

	buf.WriteString(fmt.Sprintf("Stack: ["))
	for i := 0; i < as.array.GetSize(); i++ {
		buf.WriteString(fmt.Sprintf("%v", as.array.Get(i)))
		if i != as.array.GetSize()-1 {
			buf.WriteString(", ")
		}
	}
	buf.WriteString("]")
	return buf.String()
}
